Lib/_pylong.py

```diff
@@ -0,0 +1,198 @@
+    return result
+
+
+def long_to_decimal_string(n):
+    return str(long_to_decimal(n))
+
+
+# Fast integer division, based on code from mdickinson, fast_div.py GH #47701
+
+DIV_LIMIT = 1000
+
+
+def _div2n1n(a, b, n):
+    """Divide a 2n-bit nonnegative integer a by an n-bit positive integer
+    b, using a recursive divide-and-conquer algorithm.
+
+    Inputs:
+      n is a positive integer
+      b is a positive integer with exactly n bits
+      a is a nonnegative integer such that a < 2**n * b
+
+    Output:
+      (q, r) such that a = b*q+r and 0 <= r < b.
+
+    """
+    if n <= DIV_LIMIT:
+        return divmod(a, b)
+    pad = n & 1
+    if pad:
+        a <<= 1
+        b <<= 1
+        n += 1
+    half_n = n >> 1
+    mask = (1 << half_n) - 1
+    b1, b2 = b >> half_n, b & mask
+    q1, r = _div3n2n(a >> n, (a >> half_n) & mask, b, b1, b2, half_n)
+    q2, r = _div3n2n(r, a & mask, b, b1, b2, half_n)
+    if pad:
+        r >>= 1
+    return q1 << half_n | q2, r
+
+
+def _div3n2n(a12, a3, b, b1, b2, n):
+    """Helper function for _div2n1n; not intended to be called directly."""
+    if a12 >> n == b1:
+        q, r = (1 << n) - 1, a12 - (b1 << n) + b1
+    else:
+        q, r = _div2n1n(a12, b1, n)
+    r = (r << n | a3) - q * b2
+    while r < 0:
+        q -= 1
+        r += b
+    return q, r
+
+
+def _divmod_pos(a, b):
+    """Divide a positive integer a by a positive integer b, giving
+    quotient and remainder."""
+    # Use grade-school algorithm in base 2**n, n = nbits(b)
+    n = len(bin(b)) - 2
+    mask = (1 << n) - 1
+    a_digits = []
+    while a:
+        a_digits.append(a & mask)
+        a >>= n
+    r = 0 if a_digits[-1] >= b else a_digits.pop()
+    q = 0
+    while a_digits:
+        q_digit, r = _div2n1n((r << n) + a_digits.pop(), b, n)
+        q = (q << n) + q_digit
+    return q, r
+
+
+def divmod_fast(a, b):
+    """Asymptotically fast replacement for divmod, for integers."""
+    if 0:
+        print('divmod_fast', b.bit_length(), file=sys.stderr)
+    if b == 0:
+        raise ZeroDivisionError
+    elif b < 0:
+        q, r = divmod_fast(-a, -b)
+        return q, -r
+    elif a < 0:
+        q, r = divmod_fast(~a, b)
+        return ~q, b + ~r
+    elif a == 0:
+        return 0, 0
+    else:
+        return _divmod_pos(a, b)
+
+
+# Based on code from bjorn-martinsson GH-90716, faster str-to-int conversion
+# for large numbers.
+
+@functools.cache
+def _pow5(n):
+    # Compute as needed and memoize.  Unfortunately the cache can use quite
+    # a bit of memory because the integers are large.
+    if n == 0:
+        return 5
+    p = _pow5(n-1)
+    return p*p
+
+
+def _str_to_long_inner(s):
+    def inner(l, r):
+        if r - l <= 3000:
+            return int(s[l:r])
+        lg_split = (r - l - 1).bit_length() - 1
+        split = 1 << lg_split
+        return ((inner(l, r - split) * _pow5(lg_split)) << split) + inner(
+            r - split, r
+        )
+
+    return inner(0, len(s))
```

Comment:
If this code survives, "l" has to go. From PEP 8:

> Never use the characters ‘l’ (lowercase letter el), ‘O’ (uppercase letter oh), or ‘I’ (uppercase letter eye) as single character variable names.
> In some fonts, these characters are indistinguishable from the numerals one and zero. When tempted to use ‘l’, use ‘L’ instead.

I usually use `L` and `R` in such contexts (along with `M` if I want the midpoint too).